### PR TITLE
Core/LFG: udpated and enabled CMSG_DF_GET_SYSTEM_INFO

### PR DESCRIPTION
**Pull request by Ovahlord**
**Changes proposed**:
- updated the packet structure and handling of the named opcode.

**Target branch(es)**: 6x

### CODE REVIEW

File: src/server/game/Server/Packets/LFGPackets.h [hunk 1]
```diff
@@ -31,6 +33,17 @@ namespace WorldPackets
...
+
+        class DFGetSystemInfo final : public ClientPacket
+        {
+        public:
+            DFGetSystemInfo(WorldPacket&& packet) : ClientPacket(CMSG_DF_GET_SYSTEM_INFO, std::move(packet)) { }
+
+            void Read() override;
+
+            uint8 PartyIndex = 0;
+            bool Player = 0;
```
Comment (DDuarte) [inline src/server/game/Server/Packets/LFGPackets.h]: = false;

File: src/server/game/Server/Packets/LFGPackets.h [hunk 2]
```diff
@@ -18,7 +18,9 @@
 #ifndef LFGPackets_h__
 #define LFGPackets_h__
 
+#include "Packet.h"
 #include "ObjectGuid.h"
+#include "WorldSession.h"
```
Comment (soulfrost) [inline src/server/game/Server/Packets/LFGPackets.h]: #include "WorldSession.h"
?

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 3]
```diff
@@ -17,6 +17,12 @@
 
 #include "LFGPackets.h"
 
+void WorldPackets::LFG::DFGetSystemInfo::Read()
+{
+    _worldPacket >> Player;
```
Comment (soulfrost) [inline src/server/game/Server/Packets/LFGPackets.cpp]: maybe .readbit?

File: src/server/game/Handlers/LFGHandler.cpp [hunk 4]
```diff
@@ -269,12 +270,11 @@ void WorldSession::HandleLfgTeleportOpcode(WorldPacket& recvData)
     sLFGMgr->TeleportPlayer(GetPlayer(), out, true);
 }
 
-void WorldSession::HandleDFGetSystemInfo(WorldPacket& recvData)
+void WorldSession::HandleDFGetSystemInfo(WorldPackets::LFG::DFGetSystemInfo& systemInfo)
 {
-    bool forPlayer = recvData.ReadBit();
-    TC_LOG_DEBUG("lfg", "CMSG_DF_GET_SYSTEM_INFO %s for %s", GetPlayerInfo().c_str(), (forPlayer ? "player" : "party"));
+    TC_LOG_DEBUG("lfg", "CMSG_DF_GET_SYSTEM_INFO %s for %s", GetPlayerInfo().c_str(), (systemInfo.Player ? "player" : "party"));
```
Comment (soulfrost) [inline src/server/game/Handlers/LFGHandler.cpp]: seems that logs like this uselles

File: src/server/game/Server/Protocol/Opcodes.cpp [hunk 5]
```diff
@@ -311,7 +311,7 @@ void OpcodeTable::Initialize()
     DEFINE_HANDLER(CMSG_DESTROY_ITEM,                                       STATUS_LOGGEDIN,  PROCESS_THREADUNSAFE, WorldPackets::Item::DestroyItem, &WorldSession::HandleDestroyItemOpcode);
     DEFINE_OPCODE_HANDLER_OLD(CMSG_DF_BOOT_PLAYER_VOTE,                     STATUS_UNHANDLED, PROCESS_THREADUNSAFE, &WorldSession::HandleLfgSetBootVoteOpcode      );
     DEFINE_OPCODE_HANDLER_OLD(CMSG_DF_GET_JOIN_STATUS,                      STATUS_UNHANDLED, PROCESS_THREADSAFE,   &WorldSession::HandleDFGetJoinStatus           );
-    DEFINE_OPCODE_HANDLER_OLD(CMSG_DF_GET_SYSTEM_INFO,                      STATUS_UNHANDLED, PROCESS_THREADSAFE,   &WorldSession::HandleDFGetSystemInfo           );
+    DEFINE_HANDLER(CMSG_DF_GET_SYSTEM_INFO,                                 STATUS_LOGGEDIN,  PROCESS_THREADSAFE,   WorldPackets::LFG::DFGetSystemInfo, &WorldSession::HandleDFGetSystemInfo);
```
Comment (P-Kito) [inline src/server/game/Server/Protocol/Opcodes.cpp]: Love that whitespace :P

**Pull request closed without merging**